### PR TITLE
Add support for Minio type tier

### PR DESCRIPTION
**Pull request by jinapurapu**
Note: This PR is required for [Console PR 2302](https://github.com/minio/console/pull/2302)

### CODE REVIEW

File: tier-minio.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 //
-// MinIO Object Storage (c) 2022 MinIO, Inc.
+// MinIO Object Storage (c) 2021 MinIO, Inc.
```
Comment (harshavardhana) [inline tier-minio.go:2]: it should be 2022 :-)

File: tier-minio.go [hunk 2]
```diff
@@ -18,48 +18,79 @@ package madmin
...
-	Prefix    string `json:",omitempty"`
-	Region    string `json:",omitempty"`
+	Endpoint     string `json:",omitempty"`
+	AccessKey    string `json:",omitempty"`
+	SecretKey    string `json:",omitempty"`
+	Bucket       string `json:",omitempty"`
+	Prefix       string `json:",omitempty"`
+	Region       string `json:",omitempty"`
+	StorageClass string `json:",omitempty"`
+	AWSRole      bool   `json:",omitempty"`
```
Comment (harshavardhana) [inline tier-minio.go:30]: what does AWS Role do for MinIO backend? @jinapurapu

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

File: tier-minio.go [hunk 3]
```diff
@@ -18,48 +18,70 @@ package madmin
 
 //go:generate msgp -file $GOFILE
 
-// TierMinIO represents the remote tier configuration for MinIO object storage backend.
+// TierMinIO represents the remote tier configuration for Minio type AWS S3 compatible backend.
```
Comment (dvaldivia) [inline tier-minio.go:21]: why are we changing to s3?

**Comment by krisis** [comment]
This PR is adding a new field to configure the storage-class for objects transitioned to the remote MinIO tier. The option to configure storage-class is necessary in one of the public cloud tier types since they have direct cost impact. It isn't required (or even supported) in the MinIO tier type.

**Pull request closed without merging**